src/main.rs

```diff
@@ -191,22 +214,23 @@ fn main() {
         path: PATH.to_string(),
         max_clients: MAX_CLIENTS,
         timeout: Duration::from_secs(TIMEOUT_SEC),
-        clients_count: Rc::new(RefCell::new(0u32)),
+        clients_count: Arc::new(Mutex::new(0u32)),
         timer_handle: Timer::default().handle(),
     };
     parse_opts(&mut doh);
     let listen_address = doh.listen_address;
     let listener = TcpListener::bind(&listen_address).unwrap();
     println!("Listening on http://{}", listen_address);
-    let doh = Rc::new(doh);
-    let server = Http::new()
-        .keep_alive(false)
-        .serve_incoming(listener.incoming(), move || Ok(doh.clone()));
-    let fut = server.for_each(move |client_fut| {
-        current_thread::spawn(client_fut.map(|_| {}).map_err(|_| {}));
+    let mut http = Http::new();
+    http.keep_alive(false);
+    let doh = Arc::new(Mutex::new(doh));
+    let server = listener.incoming().for_each(move |io| {
+        let service = doh.lock().unwrap().clone();
```

Comment:
Oh I forgot to mention. Previously this was only an `Rc` clone, now it's a full clone of `DoH`. The old `tokio-service` has an `impl<S: Service> Service for Rc<S>`, but hyper doesn't have it. I guess it can be added to hyper or implemented manually with something like `struct DoH { inner: Arc<DoHInner> }`.